contracts/src/utils/SendReentrancyGuard.sol

```diff
@@ -0,0 +1,40 @@
+// (c) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+// SPDX-License-Identifier: Ecosystem
+
+pragma solidity 0.8.18;
+
+/**
+ * THIS IS AN EXAMPLE CONTRACT THAT USES UN-AUDITED CODE.
+ * DO NOT USE THIS CODE IN PRODUCTION.
+ */
+
+/**
+ * @dev Abstract contract that helps implement reentrancy guards for Teleporter token bridge {_send} and {_sendAndCall}
+ * functions.
+ *
+ * The send methods must not allow reentry given that can make calls to external contracts such as {safeTransferFrom}
+ * and {safeDeposit}. However, the send methods should be allowed to be called from {receiveTeleporterMessage}, either
+ * as a part of processing a multi-hop transfer, or as a part of an external call made to process a "sendAndCall"
+ * message.
+ *
+ * @custom:security-contact https://github.com/ava-labs/teleporter-token-bridge/blob/main/SECURITY.md
+ */
+abstract contract SendReentrancyGuard {
```

Comment:
can't we directly use from `teleporter` submodule instead of copying this logic over?

Reply:
Hm, yeah good point. Though that would import that `receiverNonReentrant` guard/modifier too, which isn't used in this case (but would inflate the contract bytecode size). It could also cause confusion regarding the use of `nonReentrant` in `TeleporterUpgradable` but not using the `receiverNonReentrant` on `_receiveTeleporterMessage`. I don't feel to strongly either way TBH.

Reply:
hmm ye not a strong pref, but I'd like avoid copying this logic into different applications. I also think that it's nice to show us using Teleporter utility contracts to bring more awareness and examples for other developers. Would be nice if OZ's reentrancy guard can allow making different instances.